tests/e2e/namespace_scoped_rollouts_test.go

```diff
@@ -0,0 +1,212 @@
+		})
+
+		/*
+			In this a namespace scoped RolloutManager is created in a namespaces.
+			After creation of RMs operator should create required resources (ServiceAccount, Roles, RoleBinding, Service, Secret, Deployment) in namespace.
+			Now when a Rollouts CR is created in another namespace (which doesn't have RM), operator should not reconcile it.
+		*/
+		It("After creating namespace scoped RolloutManager in a namespace, operator should create appropriate K8s resources but it should not watch argo rollouts CR in other namespace.", func() {
+
+			nsName1, nsName2 := "test-rom-ns", "test-ro-ns"
+			testNsList = append(testNsList, nsName1, nsName2)
+			labels := map[string]string{"app": "test-argo-app"}
+
+			By("Create a namespace for rollout manager.")
+			Expect(createNamespace(ctx, k8sClient, nsName1)).To(Succeed())
+
+			By("Create namespace scoped RolloutManager in namespace.")
+			rolloutsManager, err := createRolloutManager(ctx, k8sClient, "test-rollouts-manager-1", nsName1, true)
+			Expect(err).To(Succeed())
+
+			By("Verify that RolloutManager is successfully created.")
+			Eventually(rolloutsManager, "1m", "1s").Should(rmFixture.HavePhase(rmv1alpha1.PhaseAvailable))
+
+			By("Verify that Status.Condition is set.")
+			Eventually(rolloutsManager, "1m", "1s").Should(rmFixture.HaveConditions(
+				metav1.Condition{
+					Type:    rmv1alpha1.RolloutManagerConditionTypeSuccess,
+					Status:  metav1.ConditionFalse,
+					Reason:  rmv1alpha1.RolloutManagerReasonSuccess,
+					Message: "",
+				}))
+
+			By("Verify that expected resources are created.")
+
+			validateArgoRolloutManagerResources(ctx, rolloutsManager, k8sClient, true)
+
+			By("Verify argo rollout controller is not able to reconcile CR from different namespace.")
+
+			By("Create another namespace for rollout manager.")
+			Expect(createNamespace(ctx, k8sClient, nsName2)).To(Succeed())
+
+			By("Create active and preview services in new namespace.")
+			rolloutServiceActive, err := createService(ctx, k8sClient, "rollout-bluegreen-active", nsName2, 31000, labels)
+			Expect(err).To(Succeed())
+			Eventually(&rolloutServiceActive, "10s", "1s").Should(k8s.ExistByName(k8sClient))
+
+			rolloutServicePreview, err := createService(ctx, k8sClient, "rollout-bluegreen-preview", nsName2, 32000, labels)
+			Expect(err).To(Succeed())
+			Eventually(&rolloutServicePreview, "10s", "1s").Should(k8s.ExistByName(k8sClient))
+
+			By("Create Argo Rollout CR in a different namespace and verify that it is not reconciled.")
+			rollout, err := createArgoRollout(ctx, k8sClient, "simple-rollout", nsName2, rolloutServiceActive.Name, rolloutServicePreview.Name, labels)
+			Expect(err).To(Succeed())
+			Consistently(func() bool {
+				if err := k8sClient.Get(ctx, client.ObjectKeyFromObject(&rollout), &rollout); err != nil {
+					return false
+				}
+				return reflect.DeepEqual(rollout.Status, rv1alpha1.RolloutStatus{})
+			}, "2m", "1s").Should(
```

Comment:
Waiting `2m` is a long time (it will increase the time to complete of every test run by 2 minutes), but maybe it is necessary to wait this long? Could it be shortened, or is that not feasible, here?